app/gui/view/graph-editor/src/component/visualization/container/visualization_chooser.rs

```diff
@@ -145,20 +151,12 @@ impl VisualizationChooser {
                     analytics::remote_log_value(event,field,data);
                 }
             });
+            input_type_changed <- frp.set_vis_input_type.on_change();
             frp.source.vis_input_type <+ frp.set_vis_input_type;
 
 
             // === Showing Entries ===
 
-            menu_appears <- menu.menu_visible.gate(&menu.menu_visible).constant(());
-
-            // We want to update entries according to the input type, but only when it changed and
-            // menu is visible.
-            input_type_when_visible  <- frp.set_vis_input_type.gate(&menu.menu_visible);
-            input_type_when_appeared <- frp.set_vis_input_type.sample(&menu_appears);
-            input_type               <- any(input_type_when_visible,input_type_when_appeared);
-            input_type_changed       <- input_type.on_change();
-
```

Comment:
Why this is deleted?

I'm afraid that on scenes with really many nodes we cannot afford to go through every visualization and update all chooser which are invisible anyway.

Reply:
Fair enough, but this code was essentially wrong. What happened here was, that the list of entries was only ever updated when the visualization chooser had an open selection menu. That means that the preview was never updated unless the menu was opened, and the internal state was somewhat inconsistent. 

If we want the performance benefit of not updating the menu unless shown, we need to do it in the action bar, where we have the actual information about whether the visualization chooser is visible.

Reply:
Let's go with it for now, but I'm very confused here. I wouldn't expect that updating or not the list in the _view_ would have an effect on how the visualization _updates_ work. Certainly we need to do some refactoring here.